src/main/java/bio/terra/service/dataset/flight/datadelete/DataDeletionStep.java

```diff
@@ -74,7 +86,30 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
 
   @Override
   public StepResult undoStep(FlightContext context) {
-    // The do step is atomic, either it worked or it didn't. There is no undo.
+    if (autocommit) {
+      Dataset dataset = IngestUtils.getDataset(context, datasetService);
+      UUID transactionId = TransactionUtils.getTransactionId(context);
+      DataDeletionRequest dataDeletionRequest = getRequest(context);
+      List<String> tableNames =
+          dataDeletionRequest.getTables().stream()
+              .map(DataDeletionTableModel::getTableName)
+              .collect(Collectors.toList());
+      dataset.getTables().stream()
+          .filter(t -> tableNames.contains(t.getName()))
```

Comment:
Since the only use of `tableNames` is to call `contains()`, you could build a `Set` instead of a `List` which would be more efficient.